pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -308,11 +334,49 @@ func (ctrl *PersistentVolumeController) shouldDelayBinding(claim *v1.PersistentV
 	if ctrl.isDelayBindingProvisioning(claim) {
 		return false, nil
 	}
-
 	// If claim is in delay binding mode.
 	return pvutil.IsDelayBindingMode(claim, ctrl.classLister)
 }
 
+// Unit test [12-2] [12-4]
+func (ctrl *PersistentVolumeController) createOperationTimestampCacheKey(opName, claimKey string) string {
+	return claimKey + "-" + opName
+}
+
+func (ctrl *PersistentVolumeController) recordCSIMetric(opName, claimKey string, err error) {
+	operationTsKey := ctrl.createOperationTimestampCacheKey(opName, claimKey)
+	obj, exists := ctrl.operationTimestamps.Get(operationTsKey)
+	if !exists {
+		// in this case, an entry has not been created yet by either provisionClaim or deleteVolume
+		// there are two scenarios:
+		// 1. volume already existed, thus syncUnboundClaim will *NOT* call provisionClaim
+		//    rather just bind it to a PVC, in this case, as the latency will not be reported
+		//    as it does not make much sense (will be 0). For this situation, a possible improvement
+		//    could be instead using claim's creationTimestamp in metadata, however will bring
+		//    in-consistency. Ignoring here for now
+		// 2. the plugin.IsMigratedToCSI() == false, thus provisionClaim will *NOT* create
+		//    an entry for metric recording, in this case, return directly
+		//
+		return
+	}
+	operationTs, ok := obj.(operationTimestamp)
+	if !ok {
+		klog.V(2).Infof("Cannot convert object from operationTimestamps cache to operationTimeStamp for volume claim key[%s], operation[%s]: %#v", claimKey, opName, obj)
+		return
+	}
+	// in case of error, report count instead of latency, handled by RecordVolumeOperationMetric
+	if err != nil {
```

Comment:
I am passing 0 here to avoid calling time.Since(xxx).Seconds(), also the delete cache operation only applies when a latency metric has been successfully reported

Reply:
this is now fixed